lib/vanagon/component/dsl.rb

```diff
@@ -90,7 +91,7 @@ def environment(&block)
       # @param strip [String, Integer] directory levels to skip in applying patch
       # @param fuzz [String, Integer] levels of context miss to ignore in applying patch
       def apply_patch(patch, strip: 1, fuzz: 0)
-        @component.patches << OpenStruct.new('path' => patch, 'strip' => strip.to_s, 'fuzz' => fuzz.to_s)
+        @component.patches << Vanagon::Patch.new(patch, strip, fuzz)
```

Comment:
This should probably be refactored/rebased to take PR #239 into account; `apply_patch` currently looks like this:

```ruby
def apply_patch(patch, destination: @component.dirname, strip: 1, fuzz: 0, after: 'unpack')
  raise Vanagon::Error, "We can only apply patches after the source is unpacked or after installation" unless ['unpack', 'install'].include?(after)
  @component.patches << OpenStruct.new('path' => patch, 'strip' => strip.to_s, 'fuzz' => fuzz.to_s, 'destination' => destination, 'after' => after)
end
```